### ISSUE TITLE
[Sonic Origins] Game won't open

### ISSUE THREAD

**Issue by Patrikioo** [thesupersonic16/HedgeModManager#435]
## Describe the issue/bug

Whenever I use hedge mod manager and hit save and play, the game doesn't open, but when I check the task manager the game .exe is there. When I uninstall the mod manager the game works fine and loads up normally. This also happens, when I haven't added any mods into the manager.

The debug console says this and freezes at that state:
![debug](https://user-images.githubusercontent.com/108217113/175808491-89f82644-3190-4843-b594-fda462e9ff77.JPG)

## Additional Context

-Used the newest version of hedge mod manager with both the developer version (7.8-0-b1421c6) and release-version (7.8)
-Using the Steam-Version of Sonic Origins
-Clean reinstall for both the game itself and the manager doesn't work either

**Comment by thesupersonic16** [comment]
Does the game launch when you delete `Codes.dll` from the mods folder and launch using Steam and not HMM?

**Comment by Patrikioo** [comment]
> Does the game launch when you delete `Codes.dll` from the mods folder and launch using Steam and not HMM?

Oh yes it does! Now it works, seems like the codes.dll causes some trouble. Thanks for your help! :)

**Comment by thesupersonic16** [comment]
Do you have any codes enabled?

**Comment by Patrikioo** [comment]
No, I haven't, there is only one code showing up per default it's called "Disable Auto Save Icon" but it's not checked

**Comment by thesupersonic16** [comment]
Sounds like an issue when compiling and loading the codes (even if there is none). I cannot reproduce this issue.
I'll update this issue if I get any hints on what is going on.

**Comment by CharlieSIGMA** [comment]
Same thing happens to me.
Deleting codes.dll makes game load as expected.


Sonic Origins Steam
v1.0.0.11923
![image](https://user-images.githubusercontent.com/15329810/175956769-2d894432-1b5d-4396-9947-1615a524869b.png)

Windows 10 Pro x64 21H2
Languate Portuguese-Brazil

Codes.dll attached
[Codes.zip](https://github.com/thesupersonic16/HedgeModManager/files/8992552/Codes.zip)

**Comment by thesupersonic16** [comment]
Can you try install .NET Framework 4.7.2? [Download](https://go.microsoft.com/fwlink/?linkid=863265)

**Comment by thesupersonic16** [comment]
We think the issue may be caused by a loader lock. 
This build may be buggy, but can you all try this and see if the game loads? This is not a permanent solution.

[HiteModLoader.dll](https://cdn.discordapp.com/attachments/563005172985626645/991031961885769780/HiteModLoader.dll)

**Comment by Patrikioo** [comment]
> We think the issue may be caused by a loader lock. This build may be buggy, but can you all try this and see if the game loads? This is not a permanent solution.
> 
> [HiteModLoader.dll](https://cdn.discordapp.com/attachments/563005172985626645/991031961885769780/HiteModLoader.dll)

where do I have to put the HiteModLoader.dll file?

**Comment by thesupersonic16** [comment]
Place it in the exec folder and rename it to d3d11.dll

**Comment by Patrikioo** [comment]
Ok, I did it and now the game loads up normally!

**Comment by CharlieSIGMA** [comment]
> Can you try install .NET Framework 4.7.2? [Download](https://go.microsoft.com/fwlink/?linkid=863265)

In my case I had already tried. Installer says I have a more recent version installed.

Will try the DLL later after work and report back.

**Comment by TheDraggo** [comment]
Can confirm it works if you use the provided HiteModLoader.dll

**Comment by CharlieSIGMA** [comment]
Can confirm that replacing d3d11.dll with the supplied HiteModLoader.dll works WITH codes.dll inside the MODS folder.

**Comment by thesupersonic16** [comment]
Can you try this DLL? I changed the hook and I got a single report of the issue occurring. Need to know if I broke it or not.
[HiteModLoader.zip](https://github.com/thesupersonic16/HedgeModManager/files/9003739/HiteModLoader.zip)

**Comment by slayerdude10119** [comment]
The new DLL did not work with codes.dll in the mods folder, but the previous one did, and when I update Hite, it reverts back to the previous issue of not opening.

**Comment by thesupersonic16** [comment]
This should be back to the old hook, can you confirm if this works?
[HiteModLoader.zip](https://github.com/thesupersonic16/HedgeModManager/files/9007004/HiteModLoader.zip)

**Comment by slayerdude10119** [comment]
Yup! It worked even with the codes.dll in the mods folder, so I think everything's good.

**Comment by thesupersonic16** [comment]
Since the issue seems to have been resolved, I will close this issue. If any of you have the same issue, please feel free to comment.

**Comment by jeffkrull** [comment]
I'm having the same issue; I can't get Sonic Origins to load with HMM on my Steam Deck. I'm using the Development channel of HMM (though the problem exists on Release as well), and I installed it through SteamTinkerLaunch as per the guide. It is using HiteModLoader version 1.4.0.0. I have no mods installed, and no codes downloaded, just a fresh install of Origins. If the mod loader is installed, the game will not launch. If I uninstall the mod loader, the game works. Deleting codes.dll does nothing to change this. I've also tried using the HiteModLoader.dll included in this issue, with and without codes.dll, and it still doesn't launch. I've tried enabling the debug window, but it doesn't show up when I launch the game, so I'm not sure what other debugging information I can provide.
